style/js/shield.js

```diff
@@ -219,13 +233,11 @@ export function missingIconLoader(map, e) {
   var ref = network_ref_parts[1];
   var wayName = id.split("\n")[2];
 
-  var colorLighten = ShieldDef.shieldLighten(network, ref);
-
   var ctx = drawShield(network, ref, wayName);
 
   if (ctx == null) {
     //Does not meet the criteria to draw a shield
-    return;
+    return ShieldDraw.blank();
```

Comment:
This is cleaner, but there are still early returns and an unnecessary helper function that obscures what all these code paths share in common. For future reference, what I had in mind was getting rid of early returns along these lines:

```js
function getShieldCtx(id) {
  var ctx;
  if (id !== "shield_") {
    var network_ref = …
    ctx = drawShield(network, ref, wayName);
  }
  if (ctx) {
    drawBanners(ctx, network);
    var colorLighten = ShieldDef.shieldLighten(network, ref);
    if (colorLighten) {
      let colorCtx = Gfx.getGfxContext(ctx.canvas);
      …
      ctx = colorCtx;
    }
  }
  return ctx || Gfx.getGfxContext({ width: 1, height: 1 });
}
```

This flow reduces repetition for the fallback cases without a trivial helper function. It also makes it clear that the colorLighten stuff is a better candidate for factoring out. However, this is very much in the realm of a nitpick, so unless we anticipate more edge cases and more early returns down the line, this is good enough to merge.